### PR TITLE
Adds the Sandstorm random event, directional meteor functionality, space sand.

### PR DESCRIPTION
**Pull request by Rhials**
## About The Pull Request

sandstorm

Hey guys, it's your boy. Back at it again with another meteor-adjacent event PR.

Adds the Sandstorm random event, inspired by the long-unused admin only one. It picks a direction to approach from, alerts the crew of its imminent arrival, and after a little over a minute of preparatory time, sends waves of sand and dust to grind down everything in that direction.

To accomplish this, some minor adjustments had to be made to meteor generation code. They can now be passed an optional arg for a direction to be thrown from, and will pick a random one if no direction is given.

Also introduces the newest addition to our cast of meteors -- space sand! It's even weaker than space dust, and shows up exclusively in this event. Space sand is **ineffective against rwalls**, and will not damage the arrivals area's high-tech sand-resistant glass. This is to prevent this event from venting one of the most dust-vulnerable areas on the station, and to make sure new players aren't shafted into firelock hell when the right angle is picked.

I did a lot of testing and tweaking of numbers to get the damage to average at about the level I'm comfortable with. This is meant to be a high-impact event that isn't as destructive (or unavoidable) as a meteor wave. Speaking of avoidance, let's talk about mitigation:

You get an early warning and a direction the sand will come from. You have time to grab repair supplies, move to safety, get a MODsuit. You can make worthwhile repairs as the sand comes in from inside (or outside, if you're brave enough) with nothing more than a welder and iron sheets. If you're feeling particularly spicy, you can leverage your prep time setting up shield generators, which spawn in engineering and have been added to the maintenance machines loot pool. Anyone can contribute, so do your part as a good crewmate and help out!

All that being said, the event can't be prevented entirely. Shit's going to get shredded, especially on the outside of the station. Damage will vary heavily based on the station and direction, ranging from inconsequential to threatening. It should happen late enough into the round that, at the bare minimum, the crew shouldn't be caught unprepared.

For those of you who are worried, the ORIGINAL sandstorm admin event is still with us too. It's been moved from the space dust file into the Sandstorm event file. This PR also makes a very minor change to the naming of the space _dust_ events, for better menuing.

So, to sum it all up: Sand hits grinds down one side of the station, you get a minute of warning, shield generators now spawn in maintenance. Be a good crewmate and help where you can. 

## Why It's Good For The Game

More event variety is good, and events that give the players agency on how bad the impact will be is even better.

## Changelog

:cl: Rhials
add: Sandstorm random event! A random side of the station is pummeled by an onslaught of sand and dust. If you hear that one is approaching, grab a welder and some iron to help with repairs!
add: Space sand! It's weak and doesn't hurt reinforced walls, but shouldn't be underestimated in high quantities.
code: You can now pass a start direction to the spawn_meteors/spawn_meteor global procs.
/:cl:

### CODE REVIEW

**Comment by private-tristan** [comment]
Don't shield gens require specifically science access to unlock?

**Comment by Overuseddart** [comment]
Will staff of storms be able to stop this or create a sandstorm? It does stop ash storms after all.

**Comment by ZephyrTFA** [comment]
> Will staff of storms be able to stop this or create a sandstorm? It does stop ash storms after all.

ash storms are a completely different thing

File: code/modules/meteors/meteors.dm [hunk 1]
```diff
@@ -26,33 +26,39 @@ GLOBAL_LIST_INIT(meteorsD, list(/obj/effect/meteor/medium=15, /obj/effect/meteor
 						  /obj/effect/meteor/banana=25, /obj/effect/meteor/meaty=10, /obj/effect/meteor/meaty/xeno=8, /obj/effect/meteor/emp = 30, \
 						  /obj/effect/meteor/cluster=20, /obj/effect/meteor/tunguska=1)) //for stray meteor event (bigger numbers for a bit finer weighting)
 
+GLOBAL_LIST_INIT(meteorsE, list(/obj/effect/meteor/sand=45, /obj/effect/meteor/dust=5)) //for sandstorm event
```
Comment (ZephyrTFA) [inline code/modules/meteors/meteors.dm:29]: think we could use a better name here?

Comment (ShizCalev) [reply code/modules/meteors/meteors.dm:29]: change the names of the other meteor lists too if you wouldn't mind

**Comment by moocowswag** [comment]
Anti breach shield gens dont spawn locked or anything, i use them for my maint bases alot (especially on delta) and they actually do stop otherwise highly destructive meteors.

**Review by MrMelbert — CHANGES_REQUESTED** [review]
Looks mostly good

**Pull request merged**